script.js

```diff
@@ -401,3 +401,23 @@ form.addEventListener('submit', (e) => {
     }
   }
 });
+
+// ------------------- Local-Storage ------------------
+const contactForm = document.getElementById('form');
+const Name = document.getElementById('name');
+const Email = document.getElementById('email');
+const Message = document.getElementById('message');
+
+contactForm.addEventListener('input', () => {
+  const contactData = {
+    username: Name.value,
+    email: Email.value,
+    message: Message.value,
+  };
+  localStorage.setItem('userData', JSON.stringify(contactData));
+});
+
+const GetDataFromLocalStorage = JSON.parse(localStorage.getItem('userData'));
+Name.value = GetDataFromLocalStorage.username;
+Email.value = GetDataFromLocalStorage.email;
+Message.value = GetDataFromLocalStorage.message;
```

Comment:
- Optional
Even though the code seems to work, however, there is an error in the console. ⬇️
![image](https://user-images.githubusercontent.com/33268927/202843451-a4016c8f-b567-471e-9d65-c3d6b3fa0b83.png)
The error is because when the local storage is empty, your code tries to retrieve the `userData` from the local storage. Since the local storage is empty initially, your code throws this error. You can solve this issue by using a simple `if` condition. If the `userData` exists, only then retrieve it from the local storage.